WalletWasabi.Tests/UnitTests/AllFeeEstimateTests.cs

```diff
@@ -303,7 +303,24 @@ public async Task RealWorldMempoolMinFeeAsync()
 		var estimations = feeRates.Estimations;
 		var minFee = estimations.Min(x => x.Value);
 
-		Assert.Equal(30, minFee); // this is the calculated MempoolMinFee needed to be in the top 200MB
+		Assert.Equal(20, minFee); // this is the calculated MempoolMinFee needed to be in the top 200MB
+	}
+
+	[Theory]
+	[InlineData("./UnitTests/Data/MempoolInfoWithHistogram1.json")]
+	[InlineData("./UnitTests/Data/MempoolInfoWithHistogram2.json")]
+	public async Task RealWorldMempoolRpcMinFeeAsync(string filePath)
+	{
+		var mockRpc = CreateAndConfigureRpcClient(hasPeersInfo: true);
+		var mempoolInfo = MempoolInfoGenerator.GenerateRealBitcoinKnotsMemPoolInfo(filePath);
+		mockRpc.Setup(rpc => rpc.GetMempoolInfoAsync(It.IsAny<CancellationToken>())).ReturnsAsync(mempoolInfo);
+		mockRpc.Setup(rpc => rpc.EstimateSmartFeeAsync(It.IsAny<int>(), EstimateSmartFeeMode.Conservative, It.IsAny<CancellationToken>()))
+			.ReturnsAsync(FeeRateResponse(2, 0m)); // all estimations a 0 s/b so, only estimations based on mempool.
+		var feeRates = await mockRpc.Object.EstimateAllFeeAsync(EstimateSmartFeeMode.Conservative);
+		var estimations = feeRates.Estimations;
+		var minFee = estimations.Min(x => x.Value);
+
+		Assert.Equal(2, minFee);
```

Comment:
Isn't it problematic that even with real data the minimum fee is only 2 s/vb?

Is it OK that with both Histogram the `minEstimationFor200Mb` is zero, so we go with 2 s/vb?

I know why the `minEstimationFor200Mb` is zero, because in both cases the "mempool" doesn't reach 200 Mb. 
I thought one of the histogram is from "high mempool fee spikes" days, where the mempool was way over 300 Mb.

Reply:
That was my question for you, @adamPetho. As I told you, i invested hours trying to understand why this thing didn't help us in the past.